tasks/40_find-friend/find-friend.js

```diff
@@ -0,0 +1,19 @@
+export const findFriend = (arr) => {
+  let redAmount = 0;
+
+  for (let i = 0; i < arr.length; i++) {
+    let prevElBlue = arr[i - 1] == 'blue';
+    let nextElBlue = arr[i + 1] == 'blue';
+
+    if (arr[i] !== 'red') {
+      continue;
+    } else if ((prevElBlue && nextElBlue) ||
+         (nextElBlue && arr[i + 2] == 'blue') ||
+         (prevElBlue && arr[i - 2] == 'blue')) {
+      redAmount++;
+    }
+  }
+  return redAmount;
+};
```

Comment:
This is not an algorithmic way of solving the issue, I'd recommend solving this task with the following approach: take each group in the queue (it's 3 in our case) and check this groups hats.